plugins/fastboot/fastboot.quirk

```diff
@@ -0,0 +1,4 @@
+# Google Nexus
+[DeviceInstanceId=USB\VID_18D1&PID_4EE0]
+Plugin = fastboot
+FastbootBlockSize = 512
```

Comment:
It would be nice to be able to detect compatible devices this without having to hardcode every single android device that does fastboot.

Reply:
How? We can't just do a bulk transfer on all devices with matching EPs. There are not that many fastboot devices; most if not all will use the generic Google vid:PID, even the non Google devices. I can certainly check fastboot.c and see how many vid:pids the hardcode.

Reply:
I guess matching all the VID/PIDs fastboot.c has would be fine.   I was thinking the other thing you can do is look at interface attributes to find `iInterface` matching to fastboot.

Reply:
I'm completely wrong, apologies. fastboot.c does a match on all devices with `info->ifc_class != 0xff || info->ifc_subclass != 0x42 || info->ifc_protocol != 0x03` which seems somewhat random, but consistent with things implementing fastboot and consuming it.

A am slightly worried about what happens if we just claim the fastboot interface as soon as the device is plugged in (as this PR does) -- the `fastboot` cli is going to be doing exactly the same if it's waiting for a device to appear, and I think it's fair to say that most devices in fastboot mode connected to a laptop are going to be used with fastboot rather than fwupd. Certainly whitelisting the VID/PIDs of the runtime modes might help with that, and only "connecting" to the fastboot interface if the device has the whitelisted counterpart GUID -- although this makes it somewhat fragile. Perhaps just waiting 50ms in the fwupd plugin before `claim_interface()` might be a good-enough workaround.

The other way of doing this with quirks is by defining something 100% fwupd-custom like `DeviceInstanceId=USB\IFC_CLASS_FF&IFC_SUBCLASS_42&IFC_PROTOCOL_03` although that means adding even more GUIDs to the existing devices.

Reply:
> defining something 100% fwupd-custom

Hah, turns out MS has been doing this for quite some time! See https://github.com/hughsie/fwupd/pull/866 for the new PR. These only add 3 GUIDs to devices in my system, although a lot more are calculated at startup now.

Reply:
As another thought, what about claiming the device is locked and only claim when calling unlock?

Reply:
What actually happens after setup (and of course open/close) on the device?  Does it just go back to accepting commands from other `fastboot` compatible clients?

For local development there is always `fwupdtool` which it's probably more friendly to add extra development options to interact with devices not yet marked `supported` too.